src/www/ui_tests/api/Controllers/CopyrightControllerTest.php

```diff
@@ -0,0 +1,197 @@
+   */
+  private $decisionTypes;
+
+  /**
+   * @var AssertCountBefore $assertCountBefore
+   * Assert count object
+   */
+  private $assertCountBefore;
+
+  /**
+   * @brief Setup test objects
+   * @see PHPUnit_Framework_TestCase::setUp()
+   */
+  protected function setUp(): void
+  {
+    global $container;
+    $this->userId = 2;
+    $this->groupId = 2;
+    $container = M::mock('ContainerBuilder');
+    $this->dbHelper = M::mock(DbHelper::class);
+    $this->dbManager = M::mock(DbManager::class);
+    $this->restHelper = M::mock(RestHelper::class);
+    $this->uploadDao = M::mock(UploadDao::class);
+    $this->clearingDao = M::mock(ClearingDao::class);
+    $this->CopyrightHistPlugin = M::mock(CopyrightHistogramProcessPost::class);
+    $this->decisionTypes = M::mock(DecisionTypes::class);
+    $this->licenseDao = M::mock(LicenseDao::class);
+
+    $container->shouldReceive('get')->withArgs(array(
+      'helper.restHelper'
+    ))->andReturn($this->restHelper);
+
+    $this->restHelper->shouldReceive('getPlugin')
+      ->withArgs(array('ajax-copyright-hist'))->andReturn($this->CopyrightHistPlugin);
+
+    $this->dbManager->shouldReceive('getSingleRow')
+      ->withArgs([M::any(), [
+        $this->groupId, UploadStatus::OPEN,
+        Auth::PERM_READ
+      ]]);
+    $this->dbHelper->shouldReceive('getDbManager')->andReturn($this->dbManager);
+
+    $this->restHelper->shouldReceive('getDbHelper')->andReturn($this->dbHelper);
+    $this->restHelper->shouldReceive('getGroupId')->andReturn($this->groupId);
+    $this->restHelper->shouldReceive('getUserId')->andReturn($this->userId);
+    $container->shouldReceive('get')->withArgs(['decision.types'])->andReturn($this->decisionTypes);
+    $container->shouldReceive('get')->withArgs(['dao.license'])->andReturn($this->licenseDao);
+    $container->shouldReceive('get')->withArgs(['dao.clearing'])->andReturn($this->clearingDao);
+    $container->shouldReceive('get')->withArgs(['dao.upload'])->andReturn($this->uploadDao);
+    $this->CopyrightController = new CopyrightController($container);
+    $this->assertCountBefore = \Hamcrest\MatcherAssert::getCount();
+    $this->streamFactory = new StreamFactory();
+  }
+
+  /**
+   * Helper function to get JSON array from response
+   *
+   * @param Response $response
+   * @return array Decoded response
+   */
+  private function getResponseJson($response)
+  {
+    $response->getBody()->seek(0);
+    return json_decode($response->getBody()->getContents(), true);
+  }
+
+
+  /**
+   * @test
+   * -# Test for CopyrightController::getFileCopyrights()
+   * -# Check if response status is 200 and RES body matches
+   */
+  public function testGetFileCopyrights()
+  {
+    $itemId = 68;
+    $uploadId = 4;
+    $agentId = 6;
+    $UploadTreeTableName = 'uploadtree_a';
+    $rows = array(
+      'content' => "Copyright (C) 2020 Siemens AG Author: Gaurav Mishra <mishra.gaurav@siemens.com>", 'hash' => "16f93a4e03cf6b9218fa03ca5a8ee88e",
```

Comment:
Please consider using a dummy value here instead of an actual copyright. This will result in confusion.